src/main/java/com/greenfoxacademy/opal/kalendaryo/kalendaryo/controllers/AuthContoller.java

```diff
@@ -29,12 +29,10 @@ public AuthResponse postAuth(@RequestBody AuthModel authModel, @RequestHeader("X
             userModel = new UserModel(authAndUserService.getRandomClientToken());
             userModel.setUserEmail(authModel.getEmail());
             userModel.setId(userModel.getId());
-            authAndUserService.saveUserModel(userModel);
         }
         authModel.setUser(userModel);
         authAndUserService.saveAuthModel(authModel);
-        AuthResponse authResponse = new AuthResponse(userModel.getId(), userModel.getClientToken(), authModel.getEmail(), authModel.getAccessToken());
 
-        return authResponse;
+        return new AuthResponse(userModel.getId(), userModel.getClientToken(), authModel.getEmail(), authModel.getAccessToken());
     }
 }
```

Comment:
Now I got "Column 'user_id' cannot be null" error, we have to think it over again.

Reply:
It is ok now!